pczt/src/common.rs

```diff
@@ -0,0 +1,44 @@
+/// Global fields that are relevant to the transaction as a whole.
+#[derive(Clone)]
+pub(crate) struct Global {
+    //
+    // Transaction effecting data.
+    //
+    // These are required fields that are part of the final transaction, and are filled in
+    // by the Creator when initializing the PCZT.
+    //
+    pub(crate) tx_version: u32,
+    pub(crate) version_group_id: u32,
+    /// The consensus branch ID for the chain in which this transaction will be mined.
+    ///
+    /// Non-optional because this commits to the set of consensus rules that will apply to
+    /// the transaction; differences therein can affect every role.
+    pub(crate) consensus_branch_id: u32,
```

Comment:
It could be nice to have a `network_upgrade` field instead and then infer the consensus branch id from the network upgrade, or to type this as a [`ConsensusBranchId`](https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/parameters/network_upgrade.rs#L148).

Reply:
See above. This might make sense for the Rust-specific usage (again, depending on how carefully we can manage dependencies to ensure the types are still useful to `pczt` crate consumers), but for the general PCZT format we need to think of this as being a `u32`, so I'm keeping it this way for now.